test/helpers/loans.ts

```diff
@@ -72,6 +99,34 @@ export const loanHelpers = async (
   }
 }
 
+interface Signature {
+  v: number
+  r: string
+  s: string
+}
+
+interface SignatureData {
+  signature: Signature
+  signedAt: number
+}
+
+interface LoanRequest {
+  request: LoanUserRequest
+  marketId: string
+  proof: any
+  witness: any
+  signatureData: SignatureData[]
+}
+
+interface LoanUserRequest {
+  borrower: string
+  assetAddress: string
+  assetAmount: BigNumberish
+  collateralAsset: string
+  collateralAmount: BigNumberish
+  duration: moment.Duration
+}
+
```

Comment:
instead of creating custom interfaces like this that just reflect the parameters of contract functions, you can just a handy TypeScript feature: `Parameters<ITellerDiamond['takeOutLoan']>` (`Parameters` is a built in TypeScript helper)

this way if anything changes in the contracts, we'll automatically get typing errors in TypeScript

Reply:
ouu good to know thanks